### PR TITLE
Deduplify content: Remove reviewee content from the review-guidelines

### PR DESCRIPTION
**Pull request by castrojo**
reorganize pull request page.

The review-guidelines.md had duplicate content from pull-requests.md.  I've removed it so that the review guidelines only have things useful to reviewers and pull-requests.md now has all the tips for getting a good review in one place.

The nitty gritty details of how prow/tide merges PRs is useful, but not so much when you're doing a PR, so I moved it to the bottom of the document so that the PR tips and best practices are listed before them. 

And lastly I reformatted pull-requests.md to remove the confusing numerals (they'll look better in the ToC on the contributor site this way and also reformatted it with semantic line feeds, so sorry for the noise there. 

Signed-off-by: Jorge O. Castro <jorgec@vmware.com>

**Which issue(s) this PR fixes**:

fixes: #4444

### CODE REVIEW

File: contributors/guide/pull-requests.md [hunk 1]
```diff
@@ -505,38 +476,37 @@ further help explain the _"why"_ the commit was made.
...
-## 8. KISS, YAGNI, MVP, etc.
+## KISS, YAGNI, MVP, etc.
 
 Sometimes we need to remind each other of core tenets of software design - Keep
-It Simple, You Aren't Gonna Need It, Minimum Viable Product, and so on. Adding a 
-feature "because we might need it later" is antithetical to software that ships. 
+It Simple, You Aren't Gonna Need It, Minimum Viable Product, and so on.
+Adding a feature "because we might need it later" is antithetical to software that ships. 
 Add the things you need NOW and (ideally) leave room for things you might need 
 later - but don't implement them now.
```
Comment (mrbobbytables) [inline contributors/guide/pull-requests.md:485]: Hm, what do you think of putting this and the items below above the commit msg guidelines?  I'm sort of leaning towards before it, but could go the other way too if you think thats better formatted.

File: contributors/guide/review-guidelines.md [hunk 2]
```diff
@@ -27,127 +18,10 @@ description: |
...
-
-If content and style changes were made in separate commits, they could be reviewed
-independently for their changes.
-
-**Example: Generated code**
-
-A PR that has both authored code and generated code can be difficult to parse.
-If the authored code and generated code are in separate commits, the generated
-code can largely be ignored by the reviewer, and they can focus their time on
-the authored code.
```
Comment (mrbobbytables) [inline contributors/guide/review-guidelines.md:72]: These examples might be worth integrating, they're 2 common scenarios that happen with PRs.

File: contributors/guide/review-guidelines.md [hunk 3]
```diff
@@ -27,127 +18,10 @@ description: |
...
-
-If you leave a `TODO`, it should explain why it needs to be revisited along with
-any relevant information that could help the next person looking to update that
-part of the code. Consider following insertion of a `TODO` with creating a new
-_help wanted_ issue, if appropriate.
-
-This is also true for things that should **NOT** be updated or updated with great
-care. It should document why it cannot be further reduced or changed. For an
-example of this, see the _"[KEEP THE SPACE SHUTTLE FLYING]"_ portion of the
-persistent volume controller as an example.
```
Comment (mrbobbytables) [inline contributors/guide/review-guidelines.md:98]: This should probably be integrated above as well.

**Comment by mrbobbytables** [comment]
/lgtm
/hold
For any other comments 👍 
The other parts I specifically called out should be integrated can be done in a separate PR.

**Comment by parispittman** [comment]
it would be cool if the reviewer guidelines was turned into a checklist something like
- [ ] does it have tests
- [ ] can this be broken into multiple PRs

etc

**Comment by mrbobbytables** [comment]
@parispittman there are a few items we want to add in a follow-up PR, would it be okay if those are added then?

**Comment by mrbobbytables** [comment]
discussed, we are good \o/ - changes will follow up in later PR.
/hold cancel

**Pull request merged**